.github/workflows/ci.yml

```diff
@@ -0,0 +1,54 @@
+name: nf-core CI
+# This workflow runs the pipeline with the minimal test dataset to check that it completes without any syntax errors
+on:
+  push:
+    branches:
+      - dev
+  pull_request:
+  release:
+    types: [published]
+
+jobs:
+  test:
+    name: Run workflow tests
+    # Only run on push if this is the nf-core dev branch (merged PRs)
+    if: ${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/cageseq') }}
+    runs-on: ubuntu-latest
+    env:
+      NXF_VER: ${{ matrix.nxf_ver }}
+      NXF_ANSI_LOG: false
+    strategy:
+      matrix:
+        # Nextflow versions: check pipeline minimum and current latest
+        nxf_ver: ['19.10.0', '']
+        aligner: ["--aligner bowtie1", "--aligner star"]
```

Comment:
You aren't actually using `aligner` and `options` matrices anywhere? Probably better to put these are separate tests because as it stands they will run for both `nxf_ver` which is unecessay. Have a look [here](https://github.com/nf-core/viralrecon/blob/cd031d021eea4f630260a6b8f837900779302b49/.github/workflows/ci.yml#L54).